tests/unit/wait-until-test.js

```diff
@@ -69,4 +70,19 @@ module('DOM helper: waitUntil', function () {
       assert.equal(reason.message, 'error goes here', 'valid error was thrown');
     });
   });
+
+  test('does not continue to wait if callback throws', async function (assert) {
+    assert.expect(2);
```

Comment:
For context, prior to the `waitUntil` changes above, this test failed with a large number of additional assertions (> 10+) because `waitUntil` would happily continue looping.